pkg/client/client.go

```diff
@@ -353,12 +383,22 @@ func (c *client) Get(ctx context.Context, key ObjectKey, obj Object, opts ...Get
 	}
 }
 
+// shouldReturnCacheErr determines if we should return the error we got from the cache, which we do in every case
+// except for when we're configured to do live lookups and the cache returns cache.ErrResourceNotCached
+func (c *client) shouldReturnCacheErr(err error) bool {
+	var errNotCached *cacheerrors.ErrResourceNotCached
+	return c.cacheMissPolicy != CacheMissPolicyLiveLookup || !errors.As(err, &errNotCached)
```

Comment:
Two comments:

- Can we use https://pkg.go.dev/errors#example-Is ?
- The comment explains this condition as an AND, although we're doing OR with negatives, can we flip this?

Reply:
No, you cannot use `errors.Is` as that checks for pointer equality (for e.g. a sentinel/global error value).

Yes, I can flip the boolean logic for readability.